bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BStreamType.java

```diff
@@ -175,4 +183,24 @@ protected boolean isDependentlyTypedInner(Set<MayBeDependentType> visited) {
                 (completionType instanceof MayBeDependentType completionType &&
                         completionType.isDependentlyTyped(visited));
     }
+
+
+    private static final class BStreamTypeCache {
+        record Key(int constrainId, int completionId) { }
+        record Value(int typeId, TypeCheckCache cache) { }
+        private static final LoadingCache<Key, Value> CACHE = CacheFactory.createCache(
+                BStreamTypeCache::createNewValue);
+
+        public static Value get(Type constraint, Type completion) {
+            if (constraint instanceof CacheableTypeDescriptor constraintTD &&
+                    completion instanceof CacheableTypeDescriptor completionTD) {
+                return CACHE.get(new Key(constraintTD.typeId(), completionTD.typeId()));
+            }
+            return CACHE.get(new Key(constraint.hashCode(), completion.hashCode()));
```

Comment:
Here you should use a pair of annon type ids since these hashCodes could be already allocated to types
```suggestion
            return CACHE.get(new Key(TypeIdSupplier.getAnonId(), TypeIdSupplier.getAnonId())));
```

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/43980/commits/fbbefd95988ae304196955cf3d6e84c16f330ebf